src/xbot/common/math/PID.java

```diff
@@ -102,17 +142,24 @@ public double calculate(double goal, double current,
             double p, double i, double d) {
         return calculate(goal, current, p, i, d, 0);
     }
-
+    
     /**
-     * This tells you if the controller has met its goal. Don't call this before
-     * calling calculate!
-     * 
-     * @param absoluteTolerance
-     *            How close the value can be before it is considered
-     *            "on-target."
+     * Check for all conditions where the tolerance is above 0. Since tolerances default to
+     * -1, it will skip any unassigned ones.
      */
-    public boolean isOnTarget(double absoluteTolerance)
-    {
-        return Math.abs(m_targetInputValue - m_currentInputValue) < absoluteTolerance;
+    public boolean isOnTarget() {
+        boolean checkError = errorTolerance > 0;
+        boolean checkDerivative = derivativeTolerance > 0;
+        
+        boolean isOnTarget = true;
```

Comment:
In general I don't think we'll be using these as a way to configure if a command should even finish or not. That said, I agree it's probably reasonable to say if no tolerances are set that it never finishes. 

We could preface this section with a test for any of the tolerances being there and if none return `false`?